api/client.hpp

```diff
@@ -32,6 +32,9 @@ namespace http = mender::http;
 
 using namespace std;
 
+// Inheritance here allows us to avoid re-implementing things like SetHeader(),
+// SetMethod() (and getters that come from the Request class) and we can just
+// use APIRequest instances where OutgoingRequest is needed (i.e. for HTTP).
```

Comment:
Nit: This is in the wrong commit, should be in the one that adds `APIRequest`.

Reply:
Yup, good catch! Added later so it slipped in here.